lib/administration/user/include/irods/user_administration.hpp

```diff
@@ -7,342 +7,323 @@
+    ///
+    /// \since 4.3.1
+    auto to_c_str(user_type user_type) -> const char*;
+
+    /// \brief A namespace defining the set of client-side or server-side API functions.
+    ///
+    /// This namespace's name changes based on the presence of a special macro. If the following macro
+    /// is defined, then the NAMESPACE_IMPL will be \p server, else it will be \p client.
+    ///
+    ///     - IRODS_USER_ADMINISTRATION_ENABLE_SERVER_SIDE_API
     ///
     /// \since 4.2.8
-    inline namespace v1
+    namespace NAMESPACE_IMPL
     {
-        /// \brief Defines the user types supported by the user administration library.
+        /// \brief Adds a new user to the system.
+        ///
+        /// \param[in] conn      The communication object.
+        /// \param[in] user      The user to add.
+        /// \param[in] user_type The type of the user.
+        /// \param[in] zone_type The zone that is responsible for the user.
+        ///
+        /// \throws irods::exception If an error occurs.
+        ///
+        /// \return An error code.
         ///
         /// \since 4.2.8
-        enum class user_type
-        {
-            rodsuser,   ///< Identifies users that do not have administrative privileges.
-            groupadmin, ///< Identifies users that have limited administrative privileges. 
-            rodsadmin   ///< Identifies users that have full administrative privileges.
-        }; // enum class user_type
+        auto add_user(
+            rxComm& conn,
+            const user& user,
+            user_type user_type = user_type::rodsuser,
+            zone_type zone_type = zone_type::local) -> void;
 
-        /// \brief The generic exception type used by the user administration library.
+        /// \brief Removes a user from the system.
+        ///
+        /// \param[in] conn The communication object.
+        /// \param[in] user The user to remove.
+        ///
+        /// \return An error code.
         ///
         /// \since 4.2.8
-        class user_management_error
-            : std::runtime_error
-        {
-        public:
-            using std::runtime_error::runtime_error;
-        }; // class user_management_error
+        auto remove_user(rxComm& conn, const user& user) -> void;
 
-        /// \brief Converts a string to a user_type enumeration.
+        /// \brief Changes the password of a user.
         ///
-        /// \param[in] user_type_string The string to convert.
+        /// \param[in] conn         The communication object.
+        /// \param[in] user         The user to update.
+        /// \param[in] new_password The new password of the user.
         ///
-        /// \throws user_management_error If the string cannot be converted to an enumeration.
+        /// \return An error code.
         ///
-        /// \return A user_type enumeration.
+        /// \since 4.2.8
+        auto set_user_password(rxComm& conn, const user& user, std::string_view new_password) -> void;
```

Comment:
Consider consolidating the following functions into a single function to bring it closer to other library interfaces.
- set_user_password
- set_user_type
- add_user_auth
- remove_user_auth

The new interface would follow the same pattern as the zone administration library's `modify_zone` function.

Reply:
Done.